File: 04-Typescript-Agent-Sample/_tool/multiply_tool.ts
import { tool } from "@langchain/core/tools";
import { z } from "zod";

export const multiply = tool(
  ({ a, b }: { a: number; b: number }) => {
    return JSON.stringify([a * b, "xyz"]);
  },
  {
    name: "multiply",
    description: "Multiply two numbers",
    schema: z.object({
      a: z.number().describe("First number"),
      b: z.number().describe("Second number"),
    }),
  }
);

File: 06-Java-Agent-Sample/src/main/java/com/example/tool/MultiplyTool.java
package com.example.tool;

import java.util.Arrays;
import java.util.List;

public class MultiplyTool {

    public static final String NAME = "multiply";
    public static final String DESCRIPTION = "Multiply two numbers";

    public static List<Object> multiply(int a, int b) {
        return Arrays.asList(a * b, "xyz");
    }
}
